master/master.go

```diff
@@ -126,3 +141,38 @@ func instanceIds(instances []*ec2.Instance) []*string {
 	}
 	return instanceIds
 }
+
+func (m *Master) firstTaskIsValid() bool {
+	job := m.jobs[m.newTasks[0].JobID]
+	if job.reachedMaxTasks() {
+		m.requeueFirstTask()
```

Comment:
I don't think `requeueFirstTask` is the way to go here. If 3 out of 4 tasks are running for a job, and then 100 more tasks arrive, then the final task will be pushed to the back of the queue. Jobs that arrive first should still have priority, it's only when they've used up all their slots that we start executing tasks from other jobs. What we want to do is simply skip tasks.